# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/jsonnet/hotfix-gcc16.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- a/core/unicode_test.cpp
+++ b/core/unicode_test.cpp
@@ -105,7 +105,7 @@
         size_t at = 0;
         char32_t y = decode_utf8(buffer, at);
         EXPECT_NE(y, JSONNET_CODEPOINT_ERROR) << "UTF-8 roundtrip failed for codepoint " << x << " decode rejects" << std::endl;
-        EXPECT_EQ(x, y) << "UTF-8 roundtrip failed for codepoint " << x << " converts to " << y << std::endl;
+        EXPECT_EQ(x, y) << "UTF-8 roundtrip failed for codepoint " << x << " converts to " << static_cast<unsigned int>(y) << std::endl;
         EXPECT_EQ(at, buffer.size() - 1) << "UTF-8 roundtrip failed for codepoint " << x << " decodes incorrect length" << std::endl;
     }
 }
